src/cert_store.rs

```diff
@@ -150,6 +151,51 @@ impl CertStore {
         }
     }
 
+    /// Deletes a specified store within the context of the current user.
+    /// 
+    /// Deleting stores is not natural and should only be done if necessary
+    pub fn delete_current_user_store(which: &str) -> io::Result<()> {
+        unsafe {
+            let data = OsStr::new(which)
+                .encode_wide()
+                .chain(Some(0))
+                .collect::<Vec<_>>();
+
+            // NULL is always returned when the CERT_STORE_DELETE_FLAG is set
+            let _store = Cryptography::CertOpenStore(
+                Cryptography::CERT_STORE_PROV_SYSTEM_W,
+                Cryptography::CERT_QUERY_ENCODING_TYPE::default(),
+                Cryptography::HCRYPTPROV_LEGACY::default(),
+                Cryptography::CERT_SYSTEM_STORE_CURRENT_USER_ID
+                    << Cryptography::CERT_SYSTEM_STORE_LOCATION_SHIFT | Cryptography::CERT_STORE_DELETE_FLAG,
+                data.as_ptr() as *mut _,
+            );
+            if io::Error::last_os_error().raw_os_error() != Some(0) {
+                return Err(io::Error::last_os_error());
+            }
+            else{
+                Ok(())
+            }
+        }
+    }
+
+    /// Deletes a cert as well as the cert's coresponding private key
+    /// 
+    /// This function will return an error if the provided cert does not have a private key
+    pub fn delete_cert_and_key(mut cert_context: CertContext) -> io::Result<()> {
```

Comment:
> pub fn delete_cert_and_key(mut cert_context: CertContext) -> io::Result<()> { [](http://example.com/codeflow?start=3&length=78)

Seems like this should be moved to cert_context.rs and be a function under impl CertContext. Right below where delete and delete_key_container are.